### ISSUE TITLE
vast system test suite fails

### ISSUE THREAD

**Issue by josephnoir** [vast-io/vast#141]
Executing `vast-test -s system` reports a failure:

```
[...]
+----------------------------------------------------------------------+
                                summary
+----------------------------------------------------------------------+

                        suites:  2
                        tests:   16
                        checks:  189 (188/1)
                        time:    11.29 s
                        success: 99%

                        failed tests:
                        - system:partition queries - attribute extractors

+----------------------------------------------------------------------+
```

**Comment by mavam** [comment]
Weird, I can neither reproduce this bug on macOS Sierra nor on FreeBSD. Can you provide a bit more detailed output by adding `-v 4` to the command line?

**Comment by josephnoir** [comment]
Checked out clean repositories for CAF and VAST, doesn't matter if I build with or without debug. Here is the output:

```
+----------------------------------------------------------------------+
                                 system
+----------------------------------------------------------------------+

- partition queries - attribute extractors
  -> inhaling unit test suite events [line 20]
** ../libvast/test/fixtures/events.hpp:42   !e
** ../libvast/test/fixtures/events.hpp:43   e.error() == ec::end_of_input
** ../libvast/test/fixtures/events.hpp:44   !events.empty()
** ../libvast/test/fixtures/events.hpp:42   !e
** ../libvast/test/fixtures/events.hpp:43   e.error() == ec::end_of_input
** ../libvast/test/fixtures/events.hpp:44   !events.empty()
** ../libvast/test/fixtures/events.hpp:42   !e
** ../libvast/test/fixtures/events.hpp:43   e.error() == ec::end_of_input
** ../libvast/test/fixtures/events.hpp:44   !events.empty()
** ../libvast/test/fixtures/events.hpp:42   !e
** ../libvast/test/fixtures/events.hpp:43   e.error() == ec::end_of_input
** ../libvast/test/fixtures/events.hpp:44   !events.empty()
** ../libvast/test/fixtures/events.hpp:42   !e
** ../libvast/test/fixtures/events.hpp:43   e.error() == ec::end_of_input
** ../libvast/test/fixtures/events.hpp:44   !events.empty()
  -> ingesting conn.log [line 21]
  -> ingesting http.log [line 24]
  -> ingesting bgpdump log [line 26]
  -> completed ingestion [line 28]
  -> &type [line 85]
  -> sending query [line 37]
** ../libvast/test/system/partition.cpp:39   expr
  -> shutting down partition [line 47]
** ../libvast/test/system/partition.cpp:50   exists(directory)
** ../libvast/test/system/partition.cpp:51   exists(directory / "547119946" / "data" / "id" / "orig_h")
** ../libvast/test/system/partition.cpp:52   exists(directory / "547119946" / "meta" / "time")
** ../libvast/test/system/partition.cpp:53   exists(directory / "547119946" / "meta" / "type")
  -> respawning partition and sending query again [line 54]
** ../libvast/test/system/partition.cpp:58   ::caf::test::equal_to(hits, result)
** ../libvast/test/system/partition.cpp:87   ::caf::test::equal_to(rank(hits), 4896u)
  -> sending query [line 37]
** ../libvast/test/system/partition.cpp:39   expr
  -> shutting down partition [line 47]
** ../libvast/test/system/partition.cpp:50   exists(directory)
** ../libvast/test/system/partition.cpp:51   exists(directory / "547119946" / "data" / "id" / "orig_h")
** ../libvast/test/system/partition.cpp:52   exists(directory / "547119946" / "meta" / "time")
** ../libvast/test/system/partition.cpp:53   exists(directory / "547119946" / "meta" / "type")
  -> respawning partition and sending query again [line 54]
** ../libvast/test/system/partition.cpp:58   ::caf::test::equal_to(hits, result)
** ../libvast/test/system/partition.cpp:89   ::caf::test::equal_to(rank(hits), 8462u)
  -> &time [line 90]
  -> sending query [line 37]
** ../libvast/test/system/partition.cpp:39   expr
  -> shutting down partition [line 47]
** ../libvast/test/system/partition.cpp:50   exists(directory)
** ../libvast/test/system/partition.cpp:51   exists(directory / "547119946" / "data" / "id" / "orig_h")
** ../libvast/test/system/partition.cpp:52   exists(directory / "547119946" / "meta" / "time")
** ../libvast/test/system/partition.cpp:53   exists(directory / "547119946" / "meta" / "type")
  -> respawning partition and sending query again [line 54]
** ../libvast/test/system/partition.cpp:58   ::caf::test::equal_to(hits, result)
!! ../libvast/test/system/partition.cpp:92   ::caf::test::equal_to(rank(hits), 4896u + 8462u) (0 !! 13358)
  -> 36 checks took 1.19 s
 (35/1)

+----------------------------------------------------------------------+
                                summary
+----------------------------------------------------------------------+

                        suites:  1
                        tests:   1
                        checks:  36 (35/1)
                        time:    1.19 s
                        success: 97%

                        failed tests:
                        - system:partition queries - attribute extractors

+----------------------------------------------------------------------+
```

**Comment by mavam** [comment]
(Still no luck for me in reproducing this one.)

Would you mind making available the `vast-unit-test.log` file in the build directory? This might give us more insight what's going on.

**Comment by josephnoir** [comment]
Sure, [here](https://gist.github.com/josephnoir/b9fe679381159913ac06635b8dfd5316) is the log. I have to say this is a bit unnerving. What did I do to my system? 😕

**Comment by MarcinNaw** [comment]
Your system seems just fine, I have the same issue (Arch/Antergos with GCC7)

```
    ./vast-test -s system   

    [...]
    
    +----------------------------------------------------------------------+
                                    summary
    +----------------------------------------------------------------------+
    
                            suites:  2
                            tests:   16
                            checks:  196 (195/1)
                            time:    6.47 s
                            success: 99%
    
                            failed tests:
                            - system:partition queries - attribute extractors
    
    +----------------------------------------------------------------------+

```

**Comment by MarcinNaw** [comment]
My vast-unit-test.log is available [here](https://gist.github.com/MarcinNaw/1244d898878f35af2f1787db6f79e83f).

**Comment by mavam** [comment]
Turns out that this not directly related to partition queries, but rather a bug that arises with a platform-dependent way of parsing time values: if the `TZ` environment variable is set, the unit test fails because the expression `&time > 1970-01-01` returns an non-determinstic value for the timestamp. This is because I used `std::mktime`, which makes use of TZ.

I've fixed this in branch topic/date, which is now merged into master.
